src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,104 @@
+"""Create a robot template for purpose of syncing subclassof relations.
+
+Resources
+- GitHub issue: https://github.com/monarch-initiative/mondo-ingest/issues/92
+- GitHub PR: https://github.com/monarch-initiative/mondo-ingest/pull/363
+"""
+import os
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Any, Dict, List, Union
+
+import pandas as pd
+from oaklib import get_adapter
+from oaklib.interfaces.basic_ontology_interface import RELATIONSHIP
+from oaklib.types import CURIE, URI
+
+from src.scripts.utils import DEFAULT_PREFIXES_CSV, get_monarch_curies_converter, remove_angle_brackets
+
+HERE = Path(os.path.abspath(os.path.dirname(__file__)))
+SRC_DIR = HERE.parent
+ONTOLOGY_DIR = SRC_DIR / 'ontology'
+DEFAULT_OUTPATH = ONTOLOGY_DIR / 'reports' / 'sync-subclassof.robot.template.tsv'
+DEFAULT_ONTOLOGY_DB_PATH = ONTOLOGY_DIR / 'tmp' / 'merged.db'
+ROBOT_ROW = [{
+    'term_id': 'ID',
+    'parent_id': 'SC % SPLIT=|',
+    # todo: PR comment to remind about better property for:
+    'xref_ids': 'A oboInOwl:hasDbXref SPLIT=|',  # todo: Nico may provide different value
+}]
+
+# todo: oak issue: too slow iterating. 36 seconds for below snippet
+# for mondo_id in mondo_ids[0:10]:
+#     rels: List[CURIE] = [x for x in oi.relationships(subjects=[mondo_id])]
+
+# todo: add caching?
+def sync_subclassof(
+    outpath: str = DEFAULT_OUTPATH, prefixes_csv_path: str = DEFAULT_PREFIXES_CSV,
+    ontology_db_path: str = DEFAULT_ONTOLOGY_DB_PATH
+):
+    """Run"""
+    # Vars
+    conv = get_monarch_curies_converter(prefixes_csv_path)
+    compress = lambda x: conv.compress(x) if conv.compress(x) else x
+    oi = get_adapter(ontology_db_path)
+
+    # Get all terms
+    ids_all: List[Union[CURIE, URI]] = [x for x in oi.entities(filter_obsoletes=False)]
+    mondo_ids: List[CURIE] = [x for x in ids_all if x.startswith('MONDO:')]
+
+    # todo: pr comment?: skip if parent_id not in mondo? is this only the case for: [('MONDO:0000001', 'rdfs:subClassOf', 'BFO:0000016')]
+    #  TODO: analyze first
+    #   example: mondo:0000001 rdfs:subClassOf BFO:0000016
+    # Relationships
+    # todo: PR comment: how did i implement this and why. memory vs speed. timeout on all rels. 36 seconds on iter 1x/time for just 10 records
+    # todo: oak issue: slow even batch rels (all preds). 5 seconds for 10. 25 for 100. when passed 5 preds, 1 second for 100. 36 sec for 25k~
+    rels_raw: List[RELATIONSHIP] = [x for x in oi.relationships(
+        subjects=mondo_ids,
+        # todo: pr comment: correct rels? close/related too? I think for GARD we might have 1-2 related cuz unsure
+        #  - and another question: are we indeed putting all of these in same column? (currently "'xref_ids': 'A oboInOwl:hasDbXref SPLIT=|'")
+        predicates=['rdfs:subClassOf', 'oboInOwl:hasDbXref', 'skos:exactMatch', 'skos:broadMatch', 'skos:narrowMatch']
```

Comment:
## Correct predicates?
Am I selecting the right predicates? For the [robot template](https://github.com/monarch-initiative/mondo-ingest/pull/363#discussion_r1328019845), you suggested the third column be `A oboInOwl:hasDbXref SPLIT=|`, and that you might have to think and change it.

Questions
1. I think we want more than just `oboInOwl:hasDbXref`, right?
2. And if we want to use multiple predicates, does that mean that I need to change so that have to have a separate column for each one?

Reply:
I see you explained in [this comment](https://github.com/monarch-initiative/mondo-ingest/pull/363#discussion_r1331182053). Much clearer now thanks!